plan/logical_plan_builder.go

```diff
@@ -701,9 +701,17 @@ func (b *planBuilder) buildUnion(union *ast.UnionStmt) LogicalPlan {
 
 	if union.OrderBy != nil {
 		unionPlan = b.buildSort(unionPlan, union.OrderBy.Items, nil)
+		if b.err != nil {
+			b.err = errors.Trace(b.err)
+			return nil
+		}
```

Comment:
I have a idea, should we extract this logic like this style... 

```
package main

import (
	"github.com/juju/errors"
	"fmt"
)

type builder struct {
	err error
}

func (b *builder) b1() {
	b.b2()
	b.traceSelfCallStackIfErr()
	return
}

func (b *builder) b2() {
	b.err = errors.New("111")
	return
}

func (b *builder) traceSelfCallStackIfErr() {
	if b.err == nil {
		return
	}
	err := errors.NewErrWithCause(b.err, "")
	err.SetLocation(2)
	b.err = &err
	return
}
```
and then keep all builder self call follow a `traceSelfCallStackIfErr` (we have no generic so can not wrap a function...)